lib/ansible/modules/cloud/amazon/aws_neptune_subnet_group.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Aaron Smith <ajsmith10381@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: aws_neptune_subnet_group
+short_description: Manage database subnet groups on AWS Neptune.
+description:
+    - Create, modify, and destroy database subnet groups on AWS Neptune
+version_added: "2.7"
+requirements: [ 'botocore>=1.10.0', 'boto3' ]
+author:
+    - "Aaron Smith (@slapula)"
+options:
+  name:
+    description:
+    - The name for the DB subnet group.
+    - This value is stored as a lowercase string.
+    required: true
+  state:
+    description:
+    - Whether the resource should be present or absent.
+    default: present
+    choices: ['present', 'absent']
+  description:
+    description:
+    - The description for the DB subnet group.
+    required: true
+  subnet_ids:
+    description:
+    - The EC2 Subnet IDs for the DB subnet group.
+    required: true
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = r'''
+  - name: Create a new database subnet group
+    aws_neptune_subnet_group:
+      name: "example-subnet-group"
+      state: present
+      description: 'This is an example of what you can do with this module.'
+      subnet_ids:
+        - 'subnet-1q2w3e4r'
+        - 'subnet-4r3e2w1q'
+'''
+
+RETURN = r'''
+db_subnet_group_arn:
+    description: The ARN of the database subnet group you just created or updated.
+    returned: always
+    type: string
+'''
+
+try:
+    import botocore
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.aws.waiters import get_waiter
+
+
+def group_exists(client, module, params):
+    try:
+        response = client.describe_db_subnet_groups(
+            DBSubnetGroupName=params['DBSubnetGroupName']
+        )
+    except client.exceptions.from_code('DBSubnetGroupNotFound'):
```

Comment:
Oops. Sorry. from_code does not do what I thought it did. It masks other ClientErrors. We're going to revert the places we've used this, sorry for misleading you!